### DISCUSSION TITLE
how to set Content-Encoding: gzip for BULK Queries

### DISCUSSION THREAD

**Discussion by msreekm** [jsforce/jsforce#1430] [Ideas]
In API version 27.0 and later, Bulk API can compress response data which reduces network traffic and improves response time. how do we set this option in jsforce?

Content-Encoding: gzip

**Comment by benstrum** [comment]
Somewhat related... How do you set the response to a bulk query to JSON?  Appears there is support for this: https://releasenotes.docs.salesforce.com/en-us/spring16/release-notes/rn_api_bulk.htm#rn_api_bulk_json_support.xml?edition=&impact=

**Comment by Fuchida** [comment]
@cien-ben  and anyone else who run into this, as of this comment date
converting the query output stream to a different format is not supported.

From the docs 
```
var fs = require('fs');
conn.bulk.query("SELECT Id, Name, NumberOfEmployees FROM Account")
  .stream().pipe(fs.createWriteStream('./accounts.csv'));
```

`.stream()` does not support converting from csv to other formats.

From `http://jsforce.github.io/jsforce/doc/record-stream.js.html#line85`
`    throw new Error('Converting [' + type + '] data stream is not supported.'); `

By default the records returned by `.on('record', function(rec) { console.log(rec); })`
are in JSON format.
